cfme/fixtures/v2v.py

```diff
@@ -41,6 +37,292 @@ def conversion_tags(appliance, host_creds):
     tag2 = appliance.collections.categories.instantiate(
         display_name='V2V - Transformation Method').collections.tags.instantiate(
         display_name='VDDK')
+    host_creds.remove_tags(host_creds.get_tags())
     host_creds.add_tags(tags=(tag1, tag2))
     yield
     host_creds.remove_tags(tags=(tag1, tag2))
+
+
+def get_vm(request, appliance, second_provider, template, datastore='nfs'):
+    """Fixture to provide vm object"""
+    source_datastores_list = second_provider.data.get('datastores', [False])
+    if not source_datastores_list:
+        pytest.skip("No data for source datastores in provider.")
+
+    source_datastore = [d.name for d in source_datastores_list if d.type == datastore][0]
+    collection = second_provider.appliance.provider_based_collection(second_provider)
+    vm_obj = collection.instantiate(random_vm_name('v2v'),
+                                    second_provider,
+                                    template_name=template(second_provider)['name'])
+
+    request.addfinalizer(lambda: vm_obj.cleanup_on_provider())
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default",
+         datastore=source_datastore)
+    return vm_obj
+
+
+def _form_data_cluster_mapping(second_provider, provider):
+    # since we have only one cluster on providers
```

Comment:
Two things:
1. Docblocks for these methods describing their intent.
2. Implement the patterns established in #7649 here, possibly combining all of these methods into a single one with an argument for cluster/datastore/networking, since its all the same pattern.

Reply:
What say @mshriver @kedark3?
```
def _form_data_mapping(selector, second_provider, provider, *args):
    if not selector is not 'clusters':
        source_data = second_provider.data.get(selector, [])
        target_data = provider.data.get(selector, [])

        source = [v for v in source_data if v == args[0]]
        target = [v for v in target_data if v == args[1]]

        data = not (source_data or target_data or source or target)
    else:
        source_data = second_provider.data.get(selector, [False])[0]
        target_data = provider.data.get(selector, [False])[0]
        data = not (source_data or target_data)

    if data:
        pytest.skip("No data for source or target {} in providers.".format(selector))

    return {
        'sources': [partial_match(source[0])],
        'target': [partial_match(target[0])]
    }
```

Reply:
A good start - but `source_cluster` and `target_cluster` are never returned, only `source[0]` and `target[0]`.

I think it would be better to take **kwargs and use `kwargs.get('source', '')` in place of `args[0]` or take specific kwarg names that default to None.

Reply:
I have kept these methods separate as there are some test cases that require multiple mappings for either `network` or `datastore` so I can just call required method to construct the form data more easily. I would like to keep these methods as is unless there is better way to do this which I am not seeing.

Reply:
@mshriver as we discussed on IRC I have let @Yadnyawalkya know he should be implementing code: https://github.com/ManageIQ/integration_tests/pull/7553#discussion_r211594374 in his PR #7540.